File: app/src/main/java/com/datarockets/mnchkn/activities/onboard/OnboardInteractor.java
package com.datarockets.mnchkn.activities.onboard;

public interface OnboardInteractor {

    interface OnFinishedChecking {
        void shouldShowOnboarding(boolean value);
    }

    void isUserSeenOnboarding(OnFinishedChecking listener);
    void setOnboardingSeen();
}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/onboard/OnboardInteractor.kt
package com.datarockets.mnchkn.activities.onboard

interface OnboardInteractor {

    interface OnFinishedChecking {
        fun shouldShowOnboarding(value: Boolean)
    }

    fun isUserSeenOnboarding(listener: OnFinishedChecking)
    fun setOnboardingSeen()
}
